mne/io/fiff/raw.py

```diff
@@ -115,13 +115,14 @@ def __init__(self, fname, allow_maxshield=False, preload=False,
             first_samps = list()
             last_samps = list()
             for r in raws:
-                first_samps = np.r_[first_samps, r.first_samp]
-                last_samps = np.r_[last_samps, r.last_samp]
                 self.annotations = _combine_annotations((self.annotations,
                                                          r.annotations),
                                                         last_samps,
                                                         first_samps,
-                                                        r.info['sfreq'])
+                                                        r.info['sfreq'],
+                                                        self.info['meas_date'])
+                first_samps = np.r_[first_samps, r.first_samp]
+                last_samps = np.r_[last_samps, r.last_samp]
```

Comment:
I noticed that this part of the code is not tested, but maybe I'll change that in another PR.